### PR TITLE
Change for XDG complicance

### LINKED ISSUE #1317: Follow the XDG spec for configuration
Currently the location of configuration file for mackup is `~/.mackup.cfg`, which clutters the `$HOME` directory. I believe that mackup should closely follow the [XDG specification](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html) ([Summary](https://wiki.archlinux.org/index.php/XDG_Base_Directory)) and allow users to place configuration files to `$XDG_CONFIG_HOME/mackup/config`.

### PR DESCRIPTION
**Pull request by pcr910303**
Fixes #1317  :-)

### CODE REVIEW

**Comment by pcr910303** [comment]
@lra Updated commit to be more structured, added a test for XDG config (but isn't extensively testing as the code relies on the existence of a file in disk).

**Comment by pcr910303** [comment]
@lra May I kindly ask you to review my PR? I would be very thankful if XDG landed in mackup; It's one of the five apps in my computer that doesn't follow XDG compliance.

Is there is anything that is blocking from merging?

**Comment by joshuaks** [comment]
@Ira is there any interest in merging this? Does more work need done? I'd be more than happy to get this feature in.

**Comment by lra** [comment]
Sure! I just need to protected time to get my head around it ;)

**Comment by yamatsum** [comment]
PR duplicates(https://github.com/lra/mackup/pull/632).
It looks to need `XDG_CUSTOM_APPS_DIR` not only `MACKUP_XDG_CONFIG_FILE`

**Comment by pcr910303** [comment]
@yamatsum Oops, Why did I not realize that! I should also have changed that... Maybe I'll try to resolve the conflicts, fix that problem and make a new PR tomorrow or the day after that.

**Comment by yamatsum** [comment]
@pcr910303 I think your PR is great!
The only thing I was worried about was the file name.
I would recommend the following.
```
# Mackup config file; for XDG users
XDG_MACKUP_CONFIG_FILE = 'mackup.cfg'
# Directory that can contains user defined app configs; for XDG users
XDG_CUSTOM_APPS_DIR = 'mackup'
```

**Comment by pcr910303** [comment]
OK, I'm working on this.

So I didn't want to have nested directories in `$XDG_CONFIG_HOME/mackup`, I wanted both user defined custom app configuration files and the mackup config file there. I just decided to make the config file name `config.cfg`, and when you load custom app configuration, you just ignore the file `config.cfg`.

Would this be fine?

Or... should I just revert to using a directory? 😢

**Comment by yamatsum** [comment]
@pcr910303 Thank you for a great job!
I can't judge whether the specification to implement without nesting in `$XDG_CONFIG_HOME/mackup` is good or not.
But if the file name is config.cfg, it becomes config.config.
So I think mackup.cfg is better.

File: mackup/utils.py [hunk 1]
```diff
@@ -193,6 +193,63 @@ def error(message):
...
+def get_custom_apps_dir():
+    """
+    Locate the custom apps directory.
+
+    Returns:
+        (str): Full path to the custom app directory
+    """
+    xdg_cfg_dir = get_xdg_cfg_dir()
+    home_custom_apps_dir = os.path.join(os.environ["HOME"], constants.CUSTOM_APPS_DIR)
+    if os.path.isdir(xdg_cfg_dir):
```
Comment (yamatsum) [inline mackup/utils.py:247]: If `MACKUP_XDG_CONFIG_FILE` does not exist, but `xdg_cfg_dir` does exist, is that okay?

**Comment by pcr910303** [comment]
Sorry for the late reply. I'm pretty busy, I think I won't be able to touch this PR for a while (probably until mid-June), but I just updated the PR to change it to `mackup.cfg`. I originally didn't use that because it will block making customization to mackup, but I'm not sure about that.

File: mackup/utils.py (hunk 1 above)
Comment (pcr910303) [reply mackup/utils.py:247]: I guess that would force using the default config?

**Comment by yamatsum** [comment]
@lra If you don't have any problems, can you merge?

**Comment by lra** [comment]
I started reviewing it, and was stopped, I can't blindly merge and break everyone's install, it's not as simple as this is a chicken/egg problem, the mackup config is impacting all other configs and must be loaded before anything else, but if it's in a place that depends on a config value, it won't work.

If there was a test to check this, it would be easier to review.

So I need more time to make sure this does not break, sorry.

**Comment by pcr910303** [comment]
Yeah, this PR is not really ready as @lra said. I'll try to make it ready, but it won't work in it's current state.

**Comment by pcr910303** [comment]
Ok, I'm sorry about abandoning this PR, but I clean-installed macOS Big Sur on my machine and I'm not using Mackup anymore (not that it's not useful, but I lost my Mackup backup so... 🤦🏻‍♀️), so I don't think I'll have enough time to continue this PR . Maybe someone might continue this, but I'm closing this for now.

**Comment by lra** [comment]
Done in https://github.com/lra/mackup/pull/2113 released in https://github.com/lra/mackup/releases/tag/0.9.3

**Pull request closed without merging**